src/main/java/smartthings/ratpack/sqs/internal/consumer/ConsumerManager.java

```diff
@@ -0,0 +1,88 @@
+package smartthings.ratpack.sqs.internal.consumer;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+import ratpack.exec.Execution;
+import ratpack.service.DependsOn;
+import ratpack.service.Service;
+import ratpack.service.StartEvent;
+import ratpack.service.StopEvent;
+import smartthings.ratpack.sqs.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Service responsible for managing the lifecycle of all SQS Consumer implementations defined within Registry.
+ */
+@Singleton
+@DependsOn({SqsManager.class})
+public class ConsumerManager implements Service {
+
+    private final SqsModule.Config config;
+    private final SqsManager sqsManager;
+    private List<ConsumerAction> actions = new ArrayList<>();
+    private List<CircuitBreaker> breakers = new ArrayList<>();
+
+    @Inject
+    public ConsumerManager(SqsModule.Config config, SqsManager sqsManager) {
+        this.config = config;
+        this.sqsManager = sqsManager;
+    }
+
+    @Override
+    public void onStart(StartEvent event) throws Exception {
+        if (config.isEnabled()) {
+            init(event);
+        }
+    }
+
+    @Override
+    public void onStop(StopEvent event) throws Exception {
+        actions.forEach(ConsumerAction::shutdown);
+    }
+
+    public void pause() {
+        this.breakers.forEach(CircuitBreaker::transitionToOpenState);
+    }
+
+    public void resume() {
+        this.breakers.forEach(CircuitBreaker::transitionToClosedState);
+    }
+
+    private void init(StartEvent event) {
+        this.actions = config.getConsumers().stream()
+            .flatMap(c -> buildConsumerActions(c, event))
+            .collect(Collectors.toList());
+
+        // Kick off a new execution for each defined consumer.
+        if (!this.actions.isEmpty()) {
+            this.actions.forEach((action) -> Execution.fork().start(action));
```

Comment:
Its not clear to me but I think we will lose context propagation here right?  In other words the MDC and TraceContext will not carry across to the forked execution will they?

Reply:
I suppose that the context isn't very important in the consumer case though so maybe no real value in propagating the context to the forked execution.

Reply:
That was my thought.  Our current Ratpack LoggingContext functions by binding itself to [Ratpack's RequestId](https://ratpack.io/manual/current/api/ratpack/handling/RequestId.html) which in the context of this mechanism doesn't make too much sense since this occurs outside the concept of any request.

The two thoughts I had were:
a) We could try to find a way to allow user's of library to define the context they'd like to have.  That way it isn't tied to our logging mechanisms.

b) Ignore it for now, and and consumers of the lib could potentially initialize any MDC data they need in their handlers.  I'm thinking this makes more sense to me at the moment, because that would allow the loggingId to be unique to each message being processed rather than some grouping of all messages received all time.